ssl/quic/quic_impl.c

```diff
@@ -4163,6 +4231,60 @@ SSL *ossl_quic_new_listener(SSL_CTX *ctx, uint64_t flags)
     return NULL;
 }
 
+/*
+ * SSL_new_listener_from
+ * ---------------------
+ */
+SSL *ossl_quic_new_listener_from(SSL *ssl, uint64_t flags)
+{
+    QCTX ctx;
+    QUIC_LISTENER *ql = NULL;
+    QUIC_PORT_ARGS port_args = {0};
+
+    if (!expect_quic_domain(ssl, &ctx))
+        return NULL;
+
+    qctx_lock(&ctx);
+
+    if ((ql = OPENSSL_zalloc(sizeof(*ql))) == NULL) {
+        QUIC_RAISE_NON_NORMAL_ERROR(NULL, ERR_R_CRYPTO_LIB, NULL);
+        goto err;
+    }
+
+    port_args.channel_ctx       = ssl->ctx;
+    port_args.is_multi_conn     = 1;
+    ql->port = ossl_quic_engine_create_port(ctx.qd->engine, &port_args);
+    if (ql->port == NULL) {
+        QUIC_RAISE_NON_NORMAL_ERROR(NULL, ERR_R_INTERNAL_ERROR, NULL);
+        goto err;
+    }
+
+    ql->domain  = ctx.qd;
+    ql->engine  = ctx.qd->engine;
+    ql->mutex   = ctx.qd->mutex;
```

Comment:
the definition of ql->mutex is gated on OPENSSL_THREADS.  A build configured with no-threads will break here as ->mutex will be undefined